Source/Libraries/Adapters/CsvAdapters/CsvExportAdapter.cs

```diff
@@ -356,10 +375,21 @@ public override void QueueMeasurementsForProcessing(IEnumerable<IMeasurement> me
                         measurement;
 
                     // Get last measurement timestamp -- initial timestamp is from top of second
-                    long lastTimestamp = m_lastTimestamps.GetOrDefault(timeNormalizedMeasurement.ID, _ =>
-                        timeNormalizedMeasurement.Timestamp.BaselinedTimestamp(BaselineTimeInterval.Second).Value);
-
-                    if ((timeNormalizedMeasurement.Timestamp - lastTimestamp).ToSeconds() < DownsampleInterval)
+                    if (m_lastTimestamps.TryGetValue(timeNormalizedMeasurement.ID, out long lastTimestamp))
+                    {
+                        // If last measurement timestamp is greater than current measurement timestamp, then
+                        // we have a measurement that is out of order and should be ignored
+                        if (lastTimestamp > timeNormalizedMeasurement.Timestamp.Value)
+                            continue;
+
+                        // If last measurement timestamp is within the downsampling interval, then we have
+                        // a measurement that is too close to the last measurement and should be ignored
+                        if ((timeNormalizedMeasurement.Timestamp - lastTimestamp).ToSeconds() < DownsampleInterval)
+                            continue;
+                    }
+
+                    // For normalized sub-second downsampling, only export measurements that fall on the downsampling interval
+                    if (FramesPerSecond > 0 && m_downsampleMilliseconds < 1000 & (int)timeNormalizedMeasurement.Timestamp.DistanceBeyondSecond().ToMilliseconds() % m_downsampleMilliseconds != 0)
```

Comment:
Should this be using `&&` instead of `&`?

Reply:
It seems like this logic wouldn't work for some intervals. For example, a downsample interval of 0.667 would end up behaving like a downsample interval of 1 where measurements are only exported at 667 milliseconds past the top of the second.

I think maybe what you were going for would be to compute a frame index that is equivalent to the downsampling interval at the specified framerate. Then, from a measurement timestamp, compute the number of frames since some epoch and determine if that is divisible by the frame index.